api/v1alpha1/release_webhook.go

```diff
@@ -17,38 +17,116 @@ limitations under the License.
+	mgr.GetWebhookServer().Register("/mutate-appstudio-redhat-com-v1alpha1-release", &webhook.Admission{Handler: &releaseHandler{Client: mgr.GetClient()}})
 }
 
-// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
-func (r *Release) ValidateUpdate(old runtime.Object) error {
-	if !reflect.DeepEqual(r.Spec, old.(*Release).Spec) {
-		return fmt.Errorf("release resources spec cannot be updated")
+// Handle takes an incoming admission request and returns an admission response. Create requests add
+// an author label with the current user. Update requests are rejected if the spec or author label
+// are being modified. All other requests are accepted without action.
+func (r *releaseHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	switch req.AdmissionRequest.Operation {
+	case admissionv1.Create:
+		return patchAuthorResponse(req)
+	case admissionv1.Update:
+		var releaseOld, releaseNew Release
+		err := json.Unmarshal(req.Object.Raw, &releaseNew)
+		if err != nil {
+			failedResponse(req.UID, "could not unmarshal the raw incoming Release object")
+		}
+		err = json.Unmarshal(req.OldObject.Raw, &releaseOld)
+		if err != nil {
+			failedResponse(req.UID, "could not unmarshal the raw old Release object")
+		}
+		if !reflect.DeepEqual(releaseNew.Spec, releaseOld.Spec) {
+			return failedResponse(req.UID, "release resources spec cannot be updated")
+		}
+		if releaseNew.GetLabels()[authorLabel] != releaseOld.GetLabels()[authorLabel] {
+			return failedResponse(req.UID, "release author label cannnot be updated")
+		}
 	}
+	return admission.Response{
+		AdmissionResponse: admissionv1.AdmissionResponse{
+			UID:     req.UID,
+			Allowed: true,
+			Result: &metav1.Status{
+				Status: "Success",
+			},
+		},
+	}
+}
 
-	return nil
+// patchAuthorResponse returns a failed admission.Response that blocks the requested operation.
+func failedResponse(uid types.UID, message string) admission.Response {
+	return admission.Response{
+		AdmissionResponse: admissionv1.AdmissionResponse{
+			UID:     uid,
+			Allowed: false,
+			Result: &metav1.Status{
+				Status:  "Failure",
+				Message: message,
+			},
+		},
+	}
 }
 
-// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
-func (r *Release) ValidateDelete() error {
-	return nil
+// patchAuthorResponse returns an admission.Response that modifies the CR author label to be
+// the user who issued the request.
+func patchAuthorResponse(req admission.Request) admission.Response {
+	author := strings.Replace(req.UserInfo.Username, ":", "_", -1)     // Colons disallowed in labels
+	author = strings.Replace(author, "system_serviceaccount", "sa", 1) // Effort to shorten long usernames
+	if len(author) > 63 {
+		author = string(author)[0:63] // Max length of label value is 63 chars
+	}
+	var patch []jsonpatch.JsonPatchOperation
+	patch = append(patch, jsonpatch.JsonPatchOperation{
```

Comment:
Could you use `PatchResponseFromRaw`? I think it would be cleaner. Check the test-operator webhook code.